.github/actions/setup-build-env/action.yml

```diff
@@ -0,0 +1,53 @@
+name: "Setup Dependencies Variables"
+description: "Setup Dependencies Variables from other files"
+outputs:
+  CATTLE_KDM_BRANCH:
+    description: "CATTLE_KDM_BRANCH from package/Dockerfile"
+    value: ${{ steps.vars.outputs.CATTLE_KDM_BRANCH }} 
+  CATTLE_K3S_VERSION: 
+    description: "CATTLE_K3S_VERSION from package/Dockerfile"
+    value: ${{ steps.vars.outputs.CATTLE_K3S_VERSION }}
+  CATTLE_HELM_VERSION: 
+    description: "CATTLE_HELM_VERSION from package/Dockerfile"
+    value: ${{ steps.vars.outputs.CATTLE_HELM_VERSION }}
+  HELM_VERSION: 
+    description: "HELM_VERSION from package/Dockerfile"
+    value: ${{ steps.vars.outputs.HELM_VERSION }}
+  HELM_UNITTEST_VERSION: 
+    description: "HELM_UNITTEST_VERSION"
+    value: 0.3.2
+  KUSTOMIZE_VERSION: 
+    description: "KUSTOMIZE_VERSION from package/Dockerfile"
+    value: ${{ steps.vars.outputs.KUSTOMIZE_VERSION }}
+  RKE_VERSION: 
+    description: "RKE_VERSION from go.mod"
+    value: ${{ steps.vars.outputs.RKE_VERSION }}
+  CATTLE_RANCHER_WEBHOOK_VERSION: 
+    description: "CATTLE_RANCHER_WEBHOOK_VERSION from scripts/export-config"
+    value: ${{ steps.vars.outputs.CATTLE_RANCHER_WEBHOOK_VERSION }}
+  CATTLE_CSP_ADAPTER_MIN_VERSION: 
+    description: "CATTLE_CSP_ADAPTER_MIN_VERSION from scripts/export-config"
+    value: ${{ steps.vars.outputs.CATTLE_CSP_ADAPTER_MIN_VERSION }}
+  CATTLE_FLEET_VERSION: 
+    description: "CATTLE_FLEET_VERSION from scripts/export-config"
+    value: ${{ steps.vars.outputs.CATTLE_FLEET_VERSION }}
+runs:
+  using: "composite"
+  steps:
+    - id: vars
+      shell: bash
+      run: |
+        echo "CATTLE_KDM_BRANCH=$(grep -m1 'ARG CATTLE_KDM_BRANCH=' package/Dockerfile | cut -d '=' -f2)" >> $GITHUB_OUTPUT
+        echo "CATTLE_K3S_VERSION=$(grep -m1 'ENV CATTLE_K3S_VERSION' package/Dockerfile | awk '{print $3}')" >> $GITHUB_OUTPUT
+        echo "HELM_VERSION=$(grep -m1 'ENV HELM_VERSION' package/Dockerfile | awk '{print $3}')" >> $GITHUB_OUTPUT
+        echo "KUSTOMIZE_VERSION=$(grep -m1 'ENV KUSTOMIZE_VERSION' package/Dockerfile | awk '{print $3}')" >> $GITHUB_OUTPUT
```

Comment:
Can we move this to an script instead - maybe by pulling a list of ENV/ARGs and returning a dotenv? That way we can also add some automated testing to ensure if this ever breaks, we can detect at PR time.

Reply:
I've done an alternative, that has a similar effect, checking if any variable is missing and stopping execution if there is. The downside is that we aren't checking if the values are correct